test/catalog_test.exs

```diff
@@ -34,6 +34,21 @@ defmodule Wanda.CatalogTest do
       end)
     end
 
+    test "should return the whole catalog but the check with the when clause" do
+      catalog_path = Application.fetch_env!(:wanda, Wanda.Catalog)[:catalog_path]
+
+      valid_files =
+        catalog_path
+        |> File.ls!()
+        |> Enum.sort()
+        |> Enum.filter(fn file -> file != "malformed_check.yaml" end)
+
```

Comment:
maybe we could improve this without using logic but being explicit about which checks we are expecting